privacy-on-beam/pbeam/mean_test.go

```diff
@@ -785,3 +826,128 @@ func TestFindConvertToFloat64Fn(t *testing.T) {
 		}
 	}
 }
+
+// Checks that MeanPerKey with specified partitions does cross-partition contribution bounding correctly.
+func TestMeanPerKeyWithPartitionsCrossPartitionContributionBounding(t *testing.T) {
+	var triples []tripleWithFloatValue
+
+	triples = append(triples, makeTripleWithFloatValue(1, 0, 150)...)
+	triples = append(triples, makeTripleWithFloatValue(1, 1, 150)...)
+
+	triples = append(triples, makeTripleWithFloatValueStartingFromKey(1, 50, 0, 0)...)
+	triples = append(triples, makeTripleWithFloatValueStartingFromKey(51, 50, 1, 0)...)
+
+	// MaxPartitionContributed = 1, but id = 0 contributes to 2 partitions (0 and 1).
+	// There will be cross-partition contribution bounding stage.
+	// In this stage the algorithm will typically randomly choose either contribution for partition 0 or contribution to partition 1.
+	// However, since partitions are specified, partition 1 is dropped and the algorithm will choose partition 0.
+	// The sum of 2 means should be equal to 150/51 + 0/50 = 150/51 ≈ 2.94 in both cases (unlike 150/51 + 150/51 ≈ 5.88, if no cross-partition contribution bounding is done).
+	// The difference between these numbers ≈ 2.94 and the tolerance (see below) is ≈ 0.04, so the test should catch if there was no cross-partition contribution bounding.
+	exactCount := 51.0
+	exactMean := 150.0 / exactCount
+	result := []testFloat64Metric{
+		{0, exactMean},
+	}
+	p, s, col, want := ptest.CreateList2(triples, result)
+	col = beam.ParDo(s, extractIDFromTripleWithFloatValue, col)
+
+	maxContributionsPerPartition := int64(1)
+	maxPartitionsContributed := int64(1)
+	epsilon := 60.0
+	delta := 0.01
+	lower := 0.0
+	upper := 150.0
+
+	// Specify partition 0, so id = 0 is guaranteed to contribute to partition 0.
+	partitionsCol := CreateList(s, []int{0})
```

Comment:
Again, we don't test cross-partition contribution bounding if the number of specified partitions is equal to MaxPartitionsContributed. Please fix. (Also the comments)